### PR TITLE
LanceDB Integration

### PR DESCRIPTION
**Pull request by PrashantDixit0**
This PR adds LanceDB vector store support 

- [ ] Closes #xxxx (Replace xxxx with the GitHub issue number).
- [x] Tests added and passed if fixing a bug or adding a new feature.
- [x] All [code checks passed](https://github.com/gventuri/pandas-ai/blob/main/CONTRIBUTING.md#-testing).

### CODE REVIEW

**Comment by PrashantDixit0** [comment]
@ArslanSaleem Please review this PR

File: pandasai/agent/base.py [hunk 1]
```diff
@@ -98,6 +98,19 @@ def __init__(
             self._vectorstore = BambooVectorStore(logger=self.logger)
             self.context.vectorstore = self._vectorstore
 
+        elif self._vectorstore == "lancedb" and os.environ.get("PANDASAI_API_KEY"):
```
Comment (ArslanSaleem) [inline pandasai/agent/base.py:101]: @PrashantDixit0 We don't need these lines to add lancedb check here BambooVectorStore is for default if no vectorstore is provided. The Constructor except vectorstore object not name.

File: pandasai/ee/vectorstores/lanceDB.py [hunk 2]
```diff
@@ -0,0 +1,411 @@
...
+                id: str
+                doc: str = embedding_function.SourceField()
+                metadata: str
+                vector: Vector(
+                    embedding_function.ndims()
+                ) = embedding_function.VectorField()
+
+        else:
+            # schema for qa-pairs
+            class QA_pairs(LanceModel):
```
Comment (ArslanSaleem) [inline pandasai/ee/vectorstores/lanceDB.py:67]: These classes can be declared on outside LanceDB and on the basis of check we set vector value instead. These pairs  and docs are needed for all cases.

**Review by ArslanSaleem — CHANGES_REQUESTED** [review]
@PrashantDixit0 Thanks for the changes great addition to the library!
There are just a few minor issues that need addressing. Please review the comments for details.

**Comment by PrashantDixit0** [comment]
Sure I'll make these changes asap and request again

File: pandasai/agent/base.py (hunk 1 above)
Comment (PrashantDixit0) [reply pandasai/agent/base.py]: Sure :+1:

File: pandasai/ee/vectorstores/lanceDB.py (hunk 2 above)
Comment (PrashantDixit0) [reply pandasai/ee/vectorstores/lanceDB.py]: Done

**Pull request merged**